File: frontend/src/types/preset.ts
// 预设类型
export type PresetType = 'character' | 'world' | 'style' | 'scene' | 'storyboard'

// 预设
export interface Preset {
  id: string
  owner_id?: string
  type: PresetType
  name: string
  description?: string
  data: Record<string, any>
  is_public: boolean
  created_at: string
  updated_at: string
}

// 预设列表响应
export interface PresetListResponse {
  items: Preset[]
  total: number
}

// 创建预设
export interface PresetCreate {
  type: PresetType
  name: string
  description?: string
  data: Record<string, any>
  is_public?: boolean
}

// 更新预设
export interface PresetUpdate {
  name?: string
  description?: string
  data?: Record<string, any>
  is_public?: boolean
}

// 预设类型中文映射
export const presetTypeLabels: Record<PresetType, string> = {
  character: '角色',
  world: '世界观',
  style: '风格',
  scene: '场景',
  storyboard: '分镜',
}

// URL slug 到 type 的映射
export const presetSlugToType: Record<string, PresetType> = {
  characters: 'character',
  worlds: 'world',
  styles: 'style',
  scenes: 'scene',
  storyboards: 'storyboard',
}

export const presetTypeToSlug: Record<PresetType, string> = {
  character: 'characters',
  world: 'worlds',
  style: 'styles',
  scene: 'scenes',
  storyboard: 'storyboards',
}

File: backend/app/schemas/preset.py
"""预设相关Schema"""

from datetime import datetime
from typing import Optional, Literal, Any
from pydantic import BaseModel, Field


class PresetCreate(BaseModel):
    """创建预设"""
    type: Literal["character", "world", "style", "scene", "storyboard"]
    name: str = Field(..., max_length=255)
    description: Optional[str] = None
    data: dict = {}
    is_public: bool = False


class PresetUpdate(BaseModel):
    """更新预设"""
    name: Optional[str] = Field(None, max_length=255)
    description: Optional[str] = None
    data: Optional[dict] = None
    is_public: Optional[bool] = None


class PresetResponse(BaseModel):
    """预设响应"""
    id: str
    owner_id: Optional[str] = None
    type: str
    name: str
    description: Optional[str] = None
    data: dict
    is_public: bool
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True


class PresetListResponse(BaseModel):
    """预设列表响应"""
    items: list[PresetResponse]
    total: int
